### DISCUSSION TITLE
deviceId question

### DISCUSSION THREAD

**Discussion by donniekerr** [capacitor-community/bluetooth-le#415] [Q&A]
Newbie question. 
Is the deviceId unique to each device?  i.e. if I want to related my logged in userId to their deviceId, will it stay the same and be unique?  
Is it derived from the BD_ADDR from the manufacturer? Or the MAC address I see when I look at the settings on the watch?
Just need to better understand how to the device gets a unique identity.

I'd appreciate any help.  Sorry if I shouldn't be posting questions here, but you have been very helpful to me in the past.  Thanks!

**Comment by pwespi** [comment]
Hi. I enabled the Discussions feature so that we have a place for questions / discussion and can separate them from issues.

Regarding the device ID, on Android it is indeed the Bluetooth MAC address. On iOS and web, the Bluetooth MAC address is not available (for privacy reasons). So the device ID you get is still unique, but you will get a different ID in every phone (central device) that you use.
